src/node.h

```diff
@@ -527,6 +534,23 @@ NODE_EXTERN void AddPromiseHook(v8::Isolate* isolate,
                                 promise_hook_func fn,
                                 void* arg);
 
+#if defined(__linux__)
+class AuxVector {
+ public:
+  static AuxVector& Instance() {
+    static AuxVector instance;
+    return instance;
+  }
+  std::size_t value(std::size_t type);
+ private:
+  AuxVector();
+  ~AuxVector();
+  AuxVector(AuxVector const&);
+  void operator=(AuxVector const&);
+  std::map<std::size_t, std::size_t> auxv_map_;
+};
+#endif
```

Comment:
This should not go into node.h, it's a public header.  node_internals.h is okay, though.